src/mito2/src/config.rs

```diff
@@ -298,7 +298,7 @@ impl Default for InvertedIndexConfig {
             apply_on_query: Mode::Auto,
             write_buffer_size: ReadableSize::mb(8),
             mem_threshold_on_create: Some(ReadableSize::mb(64)),
-            intermediate_path: String::new(),
+            intermediate_path: join_dir(common_config::DEFAULT_DATA_HOME, "index_intermediate"),
```

Comment:
Does it mean that we change the `intermediate_path` from `data_home/index_intermediate` to a fixed path `/tmp/greptimedb/index_intermediate`? This might be unexpected.

We should call the `MitoConfig::sanitize()` method first. It will set the intermediate path correctly.

Reply:
Maybe it's ok in `default()` ? Or should we need to sanitize the intermediate path in the `default()`? If not, we need to write the following code to make sure the config is correct:

```rust
let mut cfg = Mito::default();
cfg.sanitize("<data_home>")
``` 

Of course, the `sanitize()` needs to be exposed too.

Reply:
If we're ok with that, I can revert the above modifications and expose `sanitize()`.

Reply:
We can provide a method in `InvertedIndexConfig` to get the `intermediate_path` or return a default if it is not set. If we set the path in `default()`, then we can't distinguish whether to set the path in `sanitize()` as the path is not empty.

Reply:
It seems less elegant. The root cause is that `data_home` is not part of the Mito config. My suggestion is to make `sanitize()` public. It will be easy to initialize Mito outside the crate.

Reply:
Why dont we expose `sanitize()`?

Reply:
Feel good to me.